tests/foreman/ui/test_ansible.py

```diff
@@ -0,0 +1,356 @@
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_setup_ansible_callback_1(self):
+        """Setup Ansible callback when Ansible is installed in the Satellite
+        host
+
+        @id: d9296383-66f8-4999-9102-ef28aff77877
+
+        @assert: Ansible callback is setup succesfully. This can be verified by
+        running Ansible setup module in any of the hosts.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_setup_ansible_callback_2(self):
+        """Setup Ansible callback when Ansible is installed in the Capsule
+
+        @id: 72dec524-268e-480a-b7c1-6df2ebbe0d43
+
+        @assert: Ansible callback is setup succesfully. This can be verified by
+        running Ansible setup module in any of the hosts.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_ansible_tower_1(self):
+        """Generate Ansible run reports when Ansible Tower is installed in the
+        Satellite host
+
+        @id: 45d1f7e2-1071-432c-a6cc-7757a26ca2e4
+
+        @steps:
+
+        1. Install Ansible Tower in the Satellite host
+        2. Setup Ansible callback in the Tower
+        3. Run Ansible setup module
+
+        @assert: Ansible Tower is integrated with Satellite and the Ansible run
+        reports are shown in Satellite.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_ansible_tower_2(self):
+        """Check Ansible runs when Ansible Tower is installed in the Capsule
+
+        @id: 5af34b23-b6d7-49f5-9e92-ac0ba2182c86
+
+        @steps:
+
+        1. Install Ansible Tower in the Capsule
+        2. Setup Ansible callback in the Tower
+        3. Run Ansible setup module
+
+        @assert: Ansible Tower is integrated with Satellite and the Ansible run
+        reports are shown in Satellite.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_playbook_run_report_host_already_present(self):
+        """Ansible report creation for a host which is already present in
+        Satellite
+
+        @id: 334dfe58-6019-4d20-ab29-8cc002422ce6
+
+        @steps:
+
+        1. Register a host to Satellite
+        2. Run Ansible on the host
+
+        @assert: Ansible report is created for the host
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_playbook_run_report_host_not_present(self):
+        """Ansible report creation for a host which is not present in
+        Satellite
+
+        @id: 7f6f5767-3967-4741-9021-ce09ad69926a
+
+        @steps:
+
+        1. Run Ansible on a host not registered to Satellite yet
+
+        @assert: The host is created in Satellite and the Ansible report is
+        created for the host
+
```

Comment:
If the host is not already known to satellite (foreman), it will create a new host and create report.
If the host is already known to satellite (foreman), it will just create the report.

Hope this clarifies your question?